src/vs/platform/extensionManagement/common/extensionGalleryService.ts

```diff
@@ -882,15 +882,22 @@ abstract class AbstractExtensionGalleryService implements IExtensionGalleryServi
 			return [];
 		}
 
-		const result: IGalleryExtensionVersion[] = [];
-		for (const version of galleryExtensions[0].versions) {
-			try {
-				if (result[result.length - 1]?.version !== version.version && await this.isRawExtensionVersionCompatible(version, includePreRelease, allTargetPlatforms, targetPlatform)) {
-					result.push({ version: version.version, date: version.lastUpdated, isPreReleaseVersion: isPreReleaseVersion(version) });
-				}
-			} catch (error) { /* Ignore error and skip version */ }
-		}
-		return result;
+		const seenVersions = new Set<string>();
+		// Run checks in parallel to reduce wait time (fixes slow performance related to extensions with ownership changes)
+		const result = await Promise.all(
```

Comment:
How about use [p-map](https://www.npmjs.com/package/p-map) for control concurrently?

Reply:
I'm a new contributor to vscode. What's vscode's policy on adding more npm packages? I'd be happy to add this if it's a normal thing to do, as we can help manage how many will run at the same time.